From a2b01c8d9bd3e8f502dd96a51f8c0abbcf929559 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 29 Nov 2019 19:03:46 -0800
Subject: [PATCH] CMakeLists.txt: just depend on system fmilibrary without
 trying to build it

* -DFMILIB_BUILD_TESTS=OFF is also needed when cross-compiling

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 32 ++++++++++++++++++++------------
 1 file changed, 20 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index adb4abf..31b3468 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,19 +3,27 @@ project(fmilibrary_vendor)
 
 find_package(ament_cmake REQUIRED)
 
-include(ExternalProject)
-externalproject_add(FMILibraryProject
-  GIT_REPOSITORY https://github.com/modelon-community/fmi-library.git
-  GIT_TAG 2.1
-  TIMEOUT 60
-)
-externalproject_get_property(FMILibraryProject INSTALL_DIR)
-set(FMILibraryProject_INCLUDE_DIR "${INSTALL_DIR}/src/install/include")
-set(FMILibraryProject_LIB_DIR "${INSTALL_DIR}/src/install/lib")
+find_package(fmilibrary)
 
-install(DIRECTORY ${FMILibraryProject_INCLUDE_DIR}/ DESTINATION include)
-install(FILES ${FMILibraryProject_LIB_DIR}/libfmilib.a DESTINATION lib)
-install(FILES ${FMILibraryProject_LIB_DIR}/libfmilib_shared.so DESTINATION lib)
+# fmilibrary doesn't support CMake
+# if(NOT fmilibrary_FOUND)
+if (false)
+  include(ExternalProject)
+  externalproject_add(FMILibraryProject
+    GIT_REPOSITORY https://github.com/modelon-community/fmi-library.git
+    GIT_TAG 2.1
+    TIMEOUT 60
+  )
+  externalproject_get_property(FMILibraryProject INSTALL_DIR)
+  set(FMILibraryProject_INCLUDE_DIR "${INSTALL_DIR}/src/install/include")
+  set(FMILibraryProject_LIB_DIR "${INSTALL_DIR}/src/install/lib")
+
+  install(DIRECTORY ${FMILibraryProject_INCLUDE_DIR}/ DESTINATION include)
+  install(FILES ${FMILibraryProject_LIB_DIR}/libfmilib.a DESTINATION lib)
+  install(FILES ${FMILibraryProject_LIB_DIR}/libfmilib_shared.so DESTINATION lib)
+else()
+  message(STATUS "Found fmilibrary ${fmilibrary_VERSION}")
+endif()
 
 ament_export_include_directories(include)
 ament_export_libraries(libfmilib.a libfmilib_shared.so)
